### PR TITLE
chore(cpn) - Fix compiler warnings when building Companion.

### PR DESCRIPTION
**Pull request by philmoz**
Cleaning up some compiler warnings for Companion.

### CODE REVIEW

**Comment by pfeerick** [comment]
Need to watch out for https://github.com/EdgeTX/edgetx/pull/3339 for cleanup issues... Neil mentioned he was fixing a lot of them as part of porting to QT 5.15

File: companion/src/updates/repodatamodels.cpp [hunk 1]
```diff
@@ -138,7 +138,7 @@ bool RepoFilteredItemModel::filterAcceptsRow(int sourceRow, const QModelIndex &
   }
 }
 
-const int RepoFilteredItemModel::flags(const int id) const
+const int RepoFilteredItemModel::filterFlags(const int id) const
 {
```
Comment (elecpower) [inline companion/src/updates/repodatamodels.cpp:142]: I purposely named flags as it does not return the current datamodel m_filterflags. This function returns the processing flags for an entry. Please revert.

File: companion/src/updates/repodatamodels.h [hunk 2]
```diff
@@ -106,7 +106,7 @@ class RepoFilteredItemModel: public QSortFilterProxyModel, public RepoItemModelB
 
     const int count() const { return rowCount(); }
     const QString date(const int id) const;
-    const int flags(const int id) const;
+    const int filterFlags(const int id) const;
     const int id(const int row) const ;
```
Comment (elecpower) [inline companion/src/updates/repodatamodels.h:110]: Please revert

File: companion/src/updates/repometadata.cpp [hunk 3]
```diff
@@ -53,7 +53,7 @@ void RepoMetaData::dumpFilteredItemModel() const
 
 const int RepoMetaData::flags() const
 {
-  return m_filteredItemModel->flags(id());
+  return m_filteredItemModel->filterFlags(id());
 }
```
Comment (elecpower) [inline companion/src/updates/repometadata.cpp:57]: Please revert

**Comment by elecpower** [comment]
Make this the last one of this type until 3339 is merged. I'll need to rebase 3339 after Peter merges all the other Companion PRs as it is a growing mess so I only want to suffer that pain once but let's do it asap after 2.9.
Then after 3339 is merged my plan was to create another PR which we can add more of these compiler clean ups and move to Qt 6.
Once that is merged we will have a solid base to perform refactoring to clean up years of good intentions (mine included).

**Comment by elecpower** [comment]
Since 3339 is planned for 2.10 I'm adding as many as possible to it before merging but Qt6 and C++17 exposes even more warnings in my preliminary tests.

File: companion/src/updates/repodatamodels.cpp (hunk 1 above)
Comment (philmoz) [reply companion/src/updates/repodatamodels.cpp:142]: 'flags' clashes with a virtual function in the QSortFilterProxyModel class and generates a warning when compiling. Is there another name you would prefer?

**Comment by pfeerick** [comment]
Sounds good to me. I plan for this to be about the last PR that will touch Companion outside of trivial bugfixes before #3339 gets merged, to reduce that pain ;)

**Pull request merged**